clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy

```diff
@@ -86,24 +86,33 @@ class Keys {
       case Namespace.AZURE_SERVER_GROUPS.ns:
         def names = Names.parseName(parts[2])
         result << [
-          application: names.app.toLowerCase(),
-          name: parts[2],
-          region: parts[3],
-          account: parts[4],
-          stack: names.stack,
-          detail: names.detail
+          application : names.app.toLowerCase(),
+          serverGroup : parts[2],
+          region      : parts[3],
+          account     : parts[4],
+          stack       : names.stack,
+          detail      : names.detail
         ]
         break
       case Namespace.AZURE_CLUSTERS.ns:
-        def names = Names.parseName(parts[4])
+        def names = Names.parseName(parts[3])
         result << [
-          application: parts[3],
-          account    : parts[2],
-          name       : parts[4],
+          application: parts[2],
+          account    : parts[4],
+          name       : parts[3],
```

Comment:
why not organizing the clusters by region as well? When we run the caching agents we do run one for each region.